P3 Weather Async App

leemoispace opened the pull request:
P3 Weather Async App

Rubrics:https://learn-udacity.top/fewd586261/Front%20End%20Web%20Developer%20v5.0.0/Part%2004-Module%2001-Lesson%2005_Project%20Weather%20Journal%20App/Project%20Rubric%20-%20Weather%20Journal%20App.html


video:

https://user-images.githubusercontent.com/2882655/180152256-5c5bf0d7-3286-493d-9430-450a347ecda0.mp4

Commit:
try to update gitignore to origin, make 2 merge wrongly. this commit is the right finished version with gitignore.

Commit:
Merge branch 'main' into p3-weather-async-app

# Conflicts:
#	p3-weather-journal-app/website/app.js

solved.

Commit:
Merge branch 'main' into p3-weather-async-app

# Conflicts:
#	p3-weather-journal-app/website/app.js

Martian2Lee commented on p3-weather-journal-app/website/app.js:
```diff
@@ -3,3 +3,111 @@
 // Create a new date instance dynamically with JS
```
replace all `let` and `var` with const

leemoispace replied:
done in: [20ecb96](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/7/commits/20ecb96ba6c5ac532cbc0ea394143deb379eb132)

Martian2Lee commented on p3-weather-journal-app/website/app.js:
```diff
@@ -3,3 +3,111 @@
+};
+
+// Convert date
+function convertDate(unixtimestamp) {
```
if you copied this from else where, please attach the link to the original code

leemoispace replied:
done in: [20ecb96](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/7/commits/20ecb96ba6c5ac532cbc0ea394143deb379eb132)

Martian2Lee commented on p3-weather-journal-app/website/app.js:
```diff
@@ -3,3 +3,111 @@
+  let data = await response.json();
+  return data;
+};
+//getWeatherData(baseURL,'dubai',API_KEY)
```
remove this

leemoispace replied:
done in: [20ecb96](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/7/commits/20ecb96ba6c5ac532cbc0ea394143deb379eb132)

Martian2Lee commented on p3-weather-journal-app/website/app.js:
```diff
@@ -3,3 +3,111 @@
+  }
+};
+
+//Function to update UI
```
// Function to update UI

Martian2Lee commented on p3-weather-journal-app/website/app.js:
```diff
@@ -3,3 +3,111 @@
+    .then(() => updateUI());
+}
+
+//Function to POST data
```
// Function to POST data

Martian2Lee commented on p3-weather-journal-app/website/app.js:
```diff
@@ -3,3 +3,111 @@
 // Create a new date instance dynamically with JS
```
remove all console log

leemoispace replied:
done in: [20ecb96](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/7/commits/20ecb96ba6c5ac532cbc0ea394143deb379eb132)

Martian2Lee commented on p3-weather-journal-app/website/app.js:
```diff
@@ -3,3 +3,111 @@
+    const newData = await response.json();
+    console.log("post res: ", newData);
+  } catch (error) {
+    console.log("error", error);
```
use console.warn in catch, and do console.warn(error.message)

leemoispace replied:
done in: [20ecb96](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/7/commits/20ecb96ba6c5ac532cbc0ea394143deb379eb132)

Martian2Lee commented on p3-weather-journal-app/website/app.js:
```diff
@@ -3,3 +3,111 @@
+  let geodata = await getCityCords(cityURL, city, API_KEY);
+  //weather api: https://openweathermap.org/api/one-call-3
+  const url = `${baseURL}lat=${geodata[0].lat}&lon=${geodata[0].lon}&exclude=minutely,hourly,daily,alerts&${API_KEY}`;
+  //const response = await fetch(url);
```
remove this

leemoispace replied:
done in: [20ecb96](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/7/commits/20ecb96ba6c5ac532cbc0ea394143deb379eb132)

Martian2Lee commented on p3-weather-journal-app/website/app.js:
```diff
@@ -3,3 +3,111 @@
 // Create a new date instance dynamically with JS
```
comment consistency

leemoispace replied:
done in: [20ecb96](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/7/commits/20ecb96ba6c5ac532cbc0ea394143deb379eb132)

Martian2Lee commented on p3-weather-journal-app/website/app.js:
```diff
@@ -3,3 +3,111 @@
+      "content"
+    ).innerHTML = `Feelings: ${data.userResponse}`;
+  } catch (error) {
+    console.log("error", error);
```
use console.warn in catch, and do console.warn(error.message)

leemoispace replied:
done in: [20ecb96](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/7/commits/20ecb96ba6c5ac532cbc0ea394143deb379eb132)

Martian2Lee commented on p3-weather-journal-app/website/app.js:
```diff
@@ -3,3 +3,111 @@
 // Create a new date instance dynamically with JS
```
function order is confusing

leemoispace replied:
done in: [0648763](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/7/commits/064876345af95540b541656b9526dcd75c2822f0)

Commit:
1. Remove console.log
2. Change let var to const
3. Remove some comments and console.log
4. change console.log to warn in catches.

Commit:
change the function orders according to the calling sequence.

Martian2Lee reviewed (approved):
🚀
